// Autogenerated by metajava.py.
// Do not edit this file directly.
// The template for this file is located at:
// ../../../../../../../../templates/Exception.java
package com.rethinkdb.gen.exc;

import org.jetbrains.annotations.Nullable;
import com.rethinkdb.ast.ReqlAst;
import com.rethinkdb.model.Backtrace;

public class ReqlError extends RuntimeException {

    @Nullable Backtrace backtrace;
    @Nullable ReqlAst term;

    public ReqlError() {
    }

    public ReqlError(String message) {
        super(message);
    }

    public ReqlError(String format, Object... args) {
        super(String.format(format, args));
    }

    public ReqlError(String message, Throwable cause) {
        super(message, cause);
    }

    public ReqlError(Throwable cause) {
        super(cause);
    }

    public ReqlError(String msg, ReqlAst term, Backtrace bt) {
        super(msg);
        this.backtrace = bt;
        this.term = term;
    }

    public ReqlError setBacktrace(Backtrace backtrace) {
        this.backtrace = backtrace;
        return this;
    }

    public @Nullable Backtrace getBacktrace() {
        return backtrace;
    }

    public ReqlError setTerm(ReqlAst term) {
        this.term = term;
        return this;
    }

    public @Nullable ReqlAst getTerm() {
        return this.term;
    }
}
